File: LeetCode (en comments)/268. Missing Number.py
# 1
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        n = len(nums)
        s = int((n/2) * (1 + n))
        return s - sum(nums)

# 2
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        all_numbers = [i for i in range(len(nums) + 1)]
        return sum(all_numbers) - sum(nums)

# 3
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        all_numbers = {i for i in range(len(nums) + 1)}
        return all_numbers.difference(nums).pop()

File: LeetCode (en comments)/268. Missing Number.go
// 1
func missingNumber(nums []int) int {
    nums_len := len(nums)

    correct_nums_sum := (nums_len*(nums_len+1))/2
    sum_nums := 0

    for i := 0; i < nums_len; i++ {
        sum_nums += nums[i]
    }

    return correct_nums_sum - sum_nums
}

// 2
func missingNumber(nums []int) int {
    nums_len := len(nums)

    correct_nums_sum := 0
    sum_nums := 0

    for i := 0; i < nums_len; i++ {
        correct_nums_sum += i
        sum_nums += nums[i]
    }

    correct_nums_sum += nums_len

    return correct_nums_sum - sum_nums
}